### PR TITLE
Feature/poc/apps 139 zed module installer

### PR DESCRIPTION
**Pull request by matweew**
- Added array key support for plugin stacks:
```
"wire-plugin": [
        {
            "target": "\\Spryker\\Zed\\Payment\\PaymentDependencyProvider::getCheckoutPreCheckPlugins",
            "key": "Payone",
            "source": "\\SprykerEco\\Zed\\Payone\\Communication\\Plugin\\Checkout\\PayonePreCheckPlugin"
        },
```
- Fixed exception in `ConfigureEnvManifestStrategy.php` in case when value is an array: 
```
"configure-env": [
        {
            "target": "\\Spryker\\Shared\\Sales\\SalesConstants::PAYMENT_METHOD_STATEMACHINE_MAPPING",
            "value": {
                "payoneCreditCard": "PayoneCreditCardPartialOperations",
                "payoneInstantOnlineTransfer": "PayoneOnlineTransferPartialOperations"
            }
        }
    ],
```

### CODE REVIEW

File: src/SprykerSdk/Zed/Integrator/IntegratorConfig.php [hunk 1]
```diff
@@ -20,6 +20,7 @@ class IntegratorConfig extends AbstractBundleConfig
     public const MANIFEST_KEY_VALUE = 'value';
     public const MANIFEST_KEY_DEFAULT_VALUE = 'default';
     public const MANIFEST_KEY_CHOICES = 'choices';
+    public const MANIFEST_KEY_ARRAY_KEY = 'key';
```
Comment (matweew) [inline src/SprykerSdk/Zed/Integrator/IntegratorConfig.php:23]: probably need to find better name (eg related to value, source or target)

File: src/SprykerSdk/Zed/Integrator/Business/ManifestStrategy/CopyModuleFileManifestStrategy.php [hunk 2]
```diff
@@ -89,6 +90,6 @@ protected function getTargetPath(array $manifest): string
      */
     protected function camelCaseToDash(string $value): string
     {
-        return (new CamelCaseToSeparator('-'))->filter($value);
+        return (new UtilTextService())->camelCaseToDash($value);
```
Comment (dereuromark) [inline src/SprykerSdk/Zed/Integrator/Business/ManifestStrategy/CopyModuleFileManifestStrategy.php:93]: Binding the tool too much to modules of core will create a chicken-egg problem IMO.

**Comment by olehnovatskyi-spryker** [comment]
I guess we can close this PR because this functionality was implemented here: https://github.com/spryker-sdk/integrator/pull/14

**Comment by olehnovatskyi-spryker** [comment]
@matweew

**Comment by matweew** [comment]
Done in https://github.com/spryker-sdk/integrator/pull/14

**Pull request closed without merging**